pkg/kubelet/kubelet.go

```diff
@@ -781,8 +782,13 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		opt(klet)
 	}
 
-	klet.appArmorValidator = apparmor.NewValidator(containerRuntime)
-	klet.softAdmitHandlers.AddPodAdmitHandler(lifecycle.NewAppArmorAdmitHandler(klet.appArmorValidator))
+	if sysruntime.GOOS == "linux" {
+		// AppArmor is the linux kernel security module, and does not support non linux
+		// At present, kubelet.go rarely pays attention to the differences of os. At present, it is recommended to judge by `runtime.GOOS`,
+		// and then if there are more features, it is recommended to separate them into different files for processing.
```

Comment:
i think we can omit these comments for now.

looking at the kubelet package, there is split between using runtime checks and compile time checks (_someos.go files).
but the decisions here is to the kubelet maintainers.

Reply:
Thank you for your comment. It is a better way to split into different files according to the runtime~ But the logic is relatively simple at present, and it is currently the main entry file of kubelet, so I chose a conservative solution.

Hope to get suggestions from kubelet maintainers